docs/configuration/configuration.md

```diff
@@ -664,7 +664,15 @@ See below for the configuration options for Azure discovery:
 ```yaml
 # The information to access the Azure API.
 # The Azure environment.
-[ environment: <string> | default = AzurePublicCloud ]
+[ environment: <string|AzureCloudConfig> | default = AzurePublicCloud
```

Comment:
I would suggest to not reuse the same configuration option for two different types (I haven't seen this done anywhere else). I think a string like "custom" in environment could indicate that there are further configuration options that are taken into account.